File: app/src/main/java/com/example/pc/basemvp/screen/login/ItemUserViewModel.java
package com.example.pc.basemvp.screen.login;

import android.databinding.ObservableField;
import android.view.View;
import com.example.pc.basemvp.base.recyclerView.BaseRecyclerViewAdapter;
import com.example.pc.basemvp.data.model.User;

public class ItemUserViewModel {

    public final ObservableField<User> user = new ObservableField<>();
    private BaseRecyclerViewAdapter.OnItemClickListener<User> mItemClickListener;

    ItemUserViewModel(BaseRecyclerViewAdapter.OnItemClickListener<User> itemClickListener) {
        this.mItemClickListener = itemClickListener;
    }

    public void setData(User user) {
        this.user.set(user);
    }

    public void onItemClicked(View view) {
        if (mItemClickListener == null || this.user.get() == null) {
            return;
        }
        mItemClickListener.onItemRecyclerViewClick(user.get());
    }
}

File: app/src/main/java/com/example/mvvmkoin/screen/main/user/ItemUserViewModel.kt
package com.example.mvvmkoin.screen.main.user

import android.view.View
import androidx.databinding.Bindable
import androidx.databinding.library.baseAdapters.BR
import com.example.mvvmkoin.data.model.User
import com.example.mvvmkoin.util.extension.notNull
import com.example.mvvmkoin.widget.recyclerView.BaseItemVM

class ItemUserViewModel : BaseItemVM<User>() {

    @Bindable
    var name = ""
        set(value) {
            field = value
            notifyPropertyChanged(BR.name)
        }

    @Bindable
    var user: User? = null
        set(value) {
            field = value
            notifyPropertyChanged(BR.user)
        }

    override fun bindData(data: User?) {
        user = data
        name = position.toString() + " : " + data?.fullName
    }

    override fun onItemClicked(view: View) {
        user.notNull {
            listener?.onItemViewClick(it, position, view)
        }
    }
}